CMakeLists.txt

```diff
@@ -30,19 +30,14 @@ set(XCPP_TAGFILES_DIR ${CMAKE_CURRENT_SOURCE_DIR}/share/xeus-cpp/tagfiles)
 set(XEUS_CPP_CONF_DIR "etc/xeus-cpp"   CACHE STRING "xeus-cpp configuration directory")
 set(XCPP_TAGCONFS_DIR ${CMAKE_CURRENT_SOURCE_DIR}/etc/xeus-cpp/tags.d)
 
-configure_file("${CMAKE_CURRENT_SOURCE_DIR}/include/xeus-cpp/xeus_cpp_config.hpp.in" 
-                "${CMAKE_CURRENT_SOURCE_DIR}/include/xeus-cpp/xeus_cpp_config.hpp")
-
 # Versionning
 # ===========
 
-file(STRINGS "${XEUS_CPP_INCLUDE_DIR}/xeus-cpp/xeus_cpp_config.hpp" version_defines
-     REGEX "#define XEUS_CPP_VERSION_(MAJOR|MINOR|PATCH)")
-foreach (ver ${version_defines})
-    if (ver MATCHES "#define XEUS_CPP_VERSION_(MAJOR|MINOR|PATCH) +([^ ]+)$")
-        set(XEUS_CPP_VERSION_${CMAKE_MATCH_1} "${CMAKE_MATCH_2}" CACHE INTERNAL "")
-    endif ()
-endforeach ()
+# Set the version
+set(XEUS_CPP_VERSION_MAJOR 0)
```

Comment:
@tharun571 You probably want to set the version number the way we do in CppInterOp with a version file (see https://github.com/compiler-research/CppInterOp/pull/296/files for more details). It will allow us to label something as a version in development.

Reply:
The version file for xeus-cpp would currently have 0.5.0;dev as we are currently developing version 0.5.0 and no longer on version 0.4.0

Reply:
You won't need to add a XeusCppConfigVersion.cmake.in to be clear

Reply:
We can add a "XEUS_CPP_LABEL" or anything else to handle the "dev" part of the version (when there is one). But as said previously, I am strongly opposed to having cmake injecting the version in the cpp code.